source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+therein the test runner MUST assert that the number and sequence of expected
+events match the number and sequence of actual events observed on the specified
+client. If the list of expected events is empty, the test runner MUST assert
+that no events were observed on the client. The process for matching events is
+described in `expectedEvent`_. The test runner MUST NOT consider any
+``configureFailPoint``
+
+If `test.outcome <test_outcome_>`_ is specified, for each `collectionData`_
+therein the test runner MUST assert that the collection contains exactly the
+expected data. The test runner MUST query each collection using an ascending
+sort order on the ``_id`` field (i.e. ``{ _id: 1 }``), a ``primary`` read
+preference, a ``local`` read concern, and the internal MongoClient. If the list
+of documents is empty, the test runner MUST assert that the collection is empty.
+
+Clear the entity map for this test. For each ClientSession in the entity map,
+the test runner MUST end the session (e.g. call
+`endSession <../sessions/driver-sessions.rst#endsession>`_).
+
+If the test failed, the test runner MUST terminate any open transactions (see:
+`Terminating Open Transactions`_).
+
+Proceed to the subsequent test.
+
+
+Executing an Operation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The instructions in this section apply for each `operation`_ occuring in a
+`test`_ contained within a test file loaded by the test runner.
+
+If at any point while executing an operation an unexpected error is encountered
+or an assertion fails, the test runner MUST consider the parent test to have
+failed and proceed from `Executing a Test`_ accordingly.
+
+If `operation.object <operation_object_>`_ is "testRunner", this is a special
+operation. If `operation.name <operation_name_>`_ is defined in
+`Special Test Operations`_, the test runner MUST execute the operation
+accordingly and, if successful, proceed to the next operation in the test;
+otherwise, the test runner MUST raise an error for an undefined operation. The
+test runner MUST keep a record of any fail points configured by special
+operations so that they may be disabled after the current test.
+
+If `operation.object`_ is not "testRunner", this is an entity operation. If
+`operation.object`_ is defined in the current test's `Entity Map`_, the test
+runner MUST fetch that entity and note its type; otherwise, the test runner
+MUST raise an error for an undefined entity. If `operation.name`_ does not
+correspond to an operation for the entity type (per `Entity Test Operations`_),
+the test runner MUST raise an error for an undefined operation.
+
+Proceed with preparing the operation's arguments. If ``session`` is specified in
+`operation.arguments <operation_arguments_>`_, the test runner MUST resolve it
+to a session entity and MUST raise an error if the name is undefined or maps to
+an unexpected type. 
+
+Proceed with executing the operation such that its return value (if any) potential error can be caught
```

Comment:
nit: line too long?

Reply:
No nit too small IMO. Thanks for catching this.

My only violations of the 80-char limit should be cases where a link is too long (and in select cases where it felt more awkward to break for <5 characters of non-link text). This isn't one of those.

Reply:
Upon further review, this sentence was draft text that I forgot to remove. The paragraph that follows discusses executing the operation in a `try` block. Will just delete this.